src/hevm/src/EVM/SymExec.hs

```diff
@@ -349,33 +353,39 @@ leaves (Node _ xs) = concatMap leaves xs
 -- | Symbolically execute the VM and check all endstates against the postcondition, if available.
 -- Returns `Right (Tree BranchInfo)` if the postcondition can be violated, or
 -- or `Left (Tree BranchInfo)`, if the postcondition holds for all endstates.
-verify :: VM -> Maybe Integer -> Maybe (Fetch.BlockNumber, Text) -> Maybe Postcondition -> Query (Either (Tree BranchInfo) (Tree BranchInfo))
+verify :: VM -> Maybe Integer -> Maybe (Fetch.BlockNumber, Text) -> Maybe Postcondition -> Query VerifyResult
 verify preState maxIter rpcinfo maybepost = do
   smtState <- queryState
   tree <- doInterpret (Fetch.oracle (Just smtState) rpcinfo False) maxIter preState
   case maybepost of
     (Just post) -> do
       let livePaths = pruneDeadPaths $ leaves tree
-      -- can also do these queries individually (even concurrently!). Could save time and report multiple violations
+          -- have we hit max iterations for any path?
+          maxReached = or $ mapMaybe (flip maxIterationsReached maxIter) livePaths
```

Comment:
I don't think this will work? We will reach max iterations somewhere in the middle of a branch, but if we do we continue down to the leaf anyway, so the leaf won't remember that information. I think it might be better to add a Bool flag to `doInterpret` (and `interpret`) instead and make sure we mark max iterations there

Reply:
yes, you're right. I hadn't tested this properly :(

By mark do you just mean print a warning?

Reply:
I suppose if we could maybe add a flag to the `_iterations` state in `VM` that would let us know if max iterations had ever been reached at any point along a branch?

Reply:
or perhaps we can just check every code location in the _iterations mapping for each vm?

Reply:
I'm thinking to just pass along a bool in the `doInterpret` function would be simplest?